libs/accounts/src/lib/account-deposit.tsx

```diff
@@ -0,0 +1,142 @@
+import { forwardRef } from 'react';
+import type {
+  GroupCellRendererParams,
+  ValueFormatterParams,
+} from 'ag-grid-community';
+import { addDecimalsFormatNumber, t } from '@vegaprotocol/react-helpers';
+import { Button, Icon } from '@vegaprotocol/ui-toolkit';
+import { TooltipCellComponent } from '@vegaprotocol/ui-toolkit';
+import { AgGridDynamic as AgGrid } from '@vegaprotocol/ui-toolkit';
+import { AgGridColumn } from 'ag-grid-react';
+import type { AgGridReact, AgGridReactProps } from 'ag-grid-react';
+import type { AccountFieldsFragment } from './__generated__/Accounts';
+import { getId } from './accounts-data-provider';
+import type { AccountFields } from './accounts-manager';
+import {
+  progressBarCellRendererSelector,
+  progressBarHeaderComponentParams,
+  progressBarValueFormatter,
+} from './accounts-table';
+
+interface AccountsTableProps extends AgGridReactProps {
+  data: AccountFields[] | null;
+  onClickAsset: () => void;
+  onClickWithdraw: () => void;
+  onClickDeposit: () => void;
+  expanded: boolean;
+  showRows: boolean;
+  hideHeader?: boolean;
+}
+
+interface AccountsTableValueFormatterParams extends ValueFormatterParams {
+  data: AccountFieldsFragment;
+}
+
+export const assetDecimalsFormatter = ({
+  value,
+  data,
+}: AccountsTableValueFormatterParams) =>
+  addDecimalsFormatNumber(value, data.asset.decimals);
+
+export const AccountDeposit = forwardRef<AgGridReact, AccountsTableProps>(
```

Comment:
chore (blocking): Really not sure about this approach of rendering multiple different grids. Kind of completely defeats the purpose of using ag grid in the first place. Happy to discuss, but I think we should really just leverage what it does best, and not try to hack it so that its also an accordion

Reply:
any aggregation I was planning to use was available in the enterprise version. that is how I ended up using it this way, and I guess this is why it has been used similarly with positions